kratos/utilities/read_materials_utility.cpp

```diff
@@ -222,8 +222,36 @@ void ReadMaterialsUtility::AssignVariablesToProperty(
     // Add / override the values of material parameters in the p_properties
     if (MaterialData.Has("Variables")) {
         Parameters variables = MaterialData["Variables"];
+        
+        Parameters auxiliar_variables;
         for (auto iter = variables.begin(); iter != variables.end(); ++iter) {
             const Parameters value = variables.GetValue(iter.name());
+            std::string variable_name = iter.name();
+            TrimComponentName(variable_name);
+            // We create an axiliar parameters for adding the variables
```

Comment:
what are u doing here???

Reply:
To avoid an error when reading properties. The current implementation throws an error for non registered variables. We want to read new defined properties, not registered as variables, that the CL will store. As the first comment of this PR says:

"This PR creates an auxiliar Parameters when reading properties in order to avoid error, so these non-registered properties can be processed later"

Reply:
I still do not follow. If you try to read an unregistered variable that's by default an error.

 if you want to change this behaviour it must be in the custom reader not in the general one

Reply:
I did in the past a custom reader and it was rejected, and this was what we agreed. In any case a warning is thrown

Reply:
the point of this PR should actually exactly be about allowing a user to define __from the input__ a "custom_reader" 
(see the first file you modified).

In my opinion reading an unexisting variable should be an error not a warning (as it is now)
If you want to have a warning instead you can simply do that in the custom reader and specify in the input that you will be reading that.

Reply:
We already allowed to derive readers, it was a discussion then, and we said to do these changes. I can revert previous behaviour but keeping the code implemented here

Reply:
The current behaviour is safer for "standard" reading.
The discussion was that we should not change the mechanical_solver intrusively to do somethign special. Right now we are defining (first file changed in this PR) a way to do a custom reading elegantly, with only a minor change in the input file. 

I really see as a better solution to keep the current (safe) behaviour in the read_materials_utility.py and simply derive another class from it...

Reply:
I understand, but we have that conversation, and I think we were completely confused of the final output

Reply:
I will fix it